SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs

```diff
@@ -683,49 +684,61 @@ private static int StopDeployments(string[] args, bool useChinaPlatform)
 
             var deploymentServiceClient = DeploymentServiceClient.Create(GetApiEndpoint(useChinaPlatform), GetPlatformRefreshTokenCredential(useChinaPlatform));
 
+            var deploymentIdsToStop = new List<string>();
+
             if (args.Length == 3)
             {
                 // Stop only the specified deployment.
                 var deploymentId = args[2];
-                StopDeploymentById(deploymentServiceClient, projectName, deploymentId);
-
-                return 0;
+                deploymentIdsToStop.Add(deploymentId);
             }
-
-            // Stop all active deployments launched by this launcher.
-            var activeDeployments = ListLaunchedActiveDeployments(deploymentServiceClient, projectName);
-
-            foreach (var deployment in activeDeployments)
+            else
             {
-                var deploymentId = deployment.Id;
-                StopDeploymentById(deploymentServiceClient, projectName, deploymentId);
+                // Stop all active deployments launched by this launcher.
+                var activeDeployments = ListLaunchedActiveDeployments(deploymentServiceClient, projectName);
+                foreach( var deployment in activeDeployments)
+                {
+                    deploymentIdsToStop.Add(deployment.Id);
+                }
             }
 
-            return 0;
-        }
+            var tasks = new List<Task>();
+            var failedTasks = new List<Task>();
+
+            Console.WriteLine($"Will stop {deploymentIdsToStop.Count()} deployments");
+            foreach (var deploymentId in deploymentIdsToStop)
+            {
+                tasks.Add(StopDeploymentByIdAsync(deploymentServiceClient, projectName, deploymentId));
+            };
 
-        private static void StopDeploymentById(DeploymentServiceClient client, string projectName, string deploymentId)
-        {
             try
             {
-                Console.WriteLine($"Stopping deployment with id {deploymentId}");
-                client.StopDeployment(new StopDeploymentRequest
-                {
-                    Id = deploymentId,
-                    ProjectName = projectName
-                });
+                Task.WaitAll(tasks.ToArray());
             }
-            catch (Grpc.Core.RpcException e)
+            catch
             {
-                if (e.Status.StatusCode == Grpc.Core.StatusCode.NotFound)
+                // Retrieve individual exceptions from AggregateException thrown by Task.WaitAll
+                var throwers = tasks.Where(task => task.Exception != null);
+                foreach(Task erroredTask in throwers)
                 {
-                    Console.WriteLine("<error:unknown-deployment>");
-                }
-                else
-                {
-                    throw;
+                    Exception inner = erroredTask.Exception.InnerException;
+                    Console.WriteLine($"Error while stopping deployment: {inner.Message}");
```

Comment:
Could you also log the deployment name that had the error?

Reply:
Good point - done!